SPEC-annex-A-filesystem-storage.md

```diff
@@ -0,0 +1,49 @@
+## Annex A
+
+Annex A documents known methods of persisting GitBOM Documents to various stores.
+### GitBOM Document persistence by a Build Tool to its local filesystem
+
+If a build tool persists GitBOM information to its local filesystem, the build tool should write out the GitBOM Document to ```${GITBOM_DIR}/objects/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${GitBOMID:0:2}/${GitBOMID:2:}``` where ```${GitBOMID}``` is the GitBOM id in lowercase hexidecimal without leading zeros NOT suppressed.
+
+Example:
+
+```
+.adg/objects/gitoid/blob/sha1/0e/8efd4cdf0d5bafcfcae658c2662a73b199b301
+```
+
+#### Build tool persistence of artifact to GitBOM Document mapping
+
+A build tool should choose to persist a mapping between artifacts and their corresponding GitBOM documents.  If it chooses
+to do so it should for each artifact persist a symlink:
+
+```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${artifact id}:0:2}/${artifact id:2:} -> ${GITBOM_DIR}/objects/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${GitBOMID:0:2}/${GitBOMID:2:}```
+
+`a2g` is short for `artifact to graph`.
+
+Example:
+
+```
+.adg/a2g/gitoid/blob/sha1/0e/8efd4cdf0d5bafcfcae658c2662a73b199b301 -> .adg/objects/gitoid/blob/sha1/1d/6e79da5e380e5d3e5adcf899c4d65c0e80bfb3
+```
+
+#### Build tool persistence of related metadata
+
+A build tool may persist additional metadata to that makes reference to the Artifact Dependency Graph (ADG).
+It should persist such metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```${GITBOM_DIR}/metadata/${context}/```.  
+
+For metadata specific to a particular build tool ```${context}``` should be a name uniquely associated with the build tool.  Build tools should report their selection of ```${context}``` subdirectory name to the GitBOM spec for inclusion in a list to preclude ```${context}``` collision.
+
+Metadata persisted by multiple build tools in the same way should write a specification for that metadata.  Such specs must include the ```${context}``` for that metadata.  Such specs should be reported to the GitBOM spec for inclusion in a list to preclude ```${context}``` collision.
```

Comment:
> Metadata persisted by multiple build tools in the same way should write a specification for that metadata

This is very unclear as to who or what is writing the specification (right now it literally parses as the metadata is writing its own specification, which I guess is one unexpected advantage of welcoming the AI overlords).

Again, I'm assuming the tool developers.

Reply:
I for one welcome our robotic overlords :)

Yes, it would be the tool builders, will fix :)